### PR TITLE
TextField & TextArea: Added labelDisplay prop

### PR DESCRIPTION
**Pull request by ponori**
In order to make the `label` prop required, we need to provide a `labelDisplay` drop for toggling visibility. This work includes adding the prop to each component and providing documentation updates to each component.

### CODE REVIEW

File: docs/pages/textfield.js [hunk 1]
```diff
@@ -472,6 +498,26 @@ function Example(props) {
...
+          <Flex gap={2} direction="column">
+<Text weight="bold" size="300">First name</Text>
+<TextField
+id="textfieldexampleHiddenLabel"
+onChange={() => {}}
+label='First name'
+labelDisplay="hidden"
+size='lg'
+/>
+</Flex>
```
Comment (ayeshakmaz) [inline docs/pages/textfield.js:520]: Looks like this needs to be formatted

File: docs/pages/textarea.js [hunk 2]
```diff
@@ -104,6 +104,27 @@ function Example(props) {
...
+<Text weight="bold" size="300">About me</Text>
+<TextArea
+id="textareaexampleHiddenLabel"
+placeholder="Write something about yourself..."
+onChange={() => {}}
+label='About me'
+labelDisplay="hidden"
+size='lg'
+/>
+</Flex>
```
Comment (ayeshakmaz) [inline docs/pages/textarea.js:124]: Also needs to be formatted

File: docs/pages/searchfield.js [hunk 3]
```diff
@@ -15,7 +15,6 @@ export default function SearchFieldPage({ generatedDocGen }: {| generatedDocGen:
         defaultCode={`
   function SearchFieldExample() {
     const [value, setValue] = React.useState('');
-
     return (
```
Comment (ayeshakmaz) [inline docs/pages/searchfield.js:18]: Were all these deletions from Prettier?

File: docs/pages/searchfield.js [hunk 4]
```diff
@@ -174,7 +170,6 @@ export default function SearchFieldPage({ generatedDocGen }: {| generatedDocGen:
           title="Labels"
           description={`
       SearchField should ideally have a visible label above the input using the \`label\` prop. However, if need be, \`accessibilityLabel\` can be used to provide screen readers with context about the SearchField.
-
```
Comment (ayeshakmaz) [inline docs/pages/searchfield.js:177]: This one should probably be put back

File: docs/pages/searchfield.js [hunk 5]
```diff
@@ -223,7 +214,6 @@ export default function SearchFieldPage({ generatedDocGen }: {| generatedDocGen:
         <MainSection.Subsection
           description={`
       Be sure to localize the \`accessibilityLabel\`, \`accessibilityClearButtonLabel\`, \`errorMessage\`, \`label\` and \`placeholder\` prop values. Also localize \`value\` for those cases when it can be translated.
-
```
Comment (ayeshakmaz) [inline docs/pages/searchfield.js:226]: same here

File: docs/pages/searchfield.js [hunk 6]
```diff
@@ -382,10 +364,8 @@ function SearchFieldExample() {
           description={`
 **[ComboBox](/combobox)**
 ComboBox allows users to filter a list when selecting an option. Choose ComboBox when the user is selecting from a finite list of options.
-
```
Comment (ayeshakmaz) [inline docs/pages/searchfield.js:385]: and these two too

File: packages/gestalt/src/TextField.js [hunk 7]
```diff
@@ -38,6 +38,10 @@ type Props = {|
    * The label for the input. Be sure to localize the text.
    */
   label?: string,
+  /**
+   * Whether the legend should be visible or not. If `hidden`, the legend is still available for screen reader users, but does not appear visually. See the [label visibility variant](https://gestalt.pinterest.systems#Label-visibility) for more info.
```
Comment (ayeshakmaz) [inline packages/gestalt/src/TextField.js:42]: replace legend with label here

File: packages/gestalt/src/TextArea.js [hunk 8]
```diff
@@ -38,6 +38,10 @@ type Props = {|
    * The label for the input. Be sure to localize the text.
    */
   label?: string,
+  /**
+   * Whether the legend should be visible or not. If `hidden`, the legend is still available for screen reader users, but does not appear visually. See the [label visibility variant](https://gestalt.pinterest.systems#Label-visibility) for more info.
```
Comment (ayeshakmaz) [inline packages/gestalt/src/TextArea.js:42]: replace legend with label here

**Review by ayeshakmaz — CHANGES_REQUESTED** [review]

File: docs/pages/textfield.js (hunk 1 above)
Comment (ponori) [reply docs/pages/textfield.js:520]: Done!

File: docs/pages/textarea.js (hunk 2 above)
Comment (ponori) [reply docs/pages/textarea.js:124]: Done!

File: docs/pages/searchfield.js (hunk 3 above)
Comment (ponori) [reply docs/pages/searchfield.js:18]: I believe so.

File: packages/gestalt/src/TextArea.js (hunk 8 above)
Comment (ponori) [reply packages/gestalt/src/TextArea.js:42]: Done!

File: packages/gestalt/src/TextField.js (hunk 7 above)
Comment (ponori) [reply packages/gestalt/src/TextField.js:42]: Done!

File: docs/pages/searchfield.js (hunk 3 above)
Comment (ponori) [reply docs/pages/searchfield.js:18]: Added back.

File: docs/pages/searchfield.js (hunk 4 above)
Comment (ponori) [reply docs/pages/searchfield.js:177]: Done!

File: docs/pages/searchfield.js (hunk 5 above)
Comment (ponori) [reply docs/pages/searchfield.js:226]: Done!

**Review by ayeshakmaz — APPROVED** [review]
The line breaks in the Related section of SearchField still need to be added back, but otherwise looks good!

File: docs/pages/searchfield.js [hunk 9]
```diff
@@ -24,7 +24,6 @@ export default function SearchFieldPage({ generatedDocGen }: {| generatedDocGen:
           size={20}
           accessibilityLabel="Pinterest"
         />
-
```
Comment (dangerismycat) [inline docs/pages/searchfield.js:27]: What are all these unrelated whitespace changes doing in this PR?
